app/scripts/lib/transaction/smart-transactions.ts

```diff
@@ -210,14 +216,129 @@ class SmartTransactionHook {
     }
   }
 
+  async submitBatch() {
+    // Will cause TransactionController to publish to the RPC provider as normal.
+    const useRegularTransactionSubmit = undefined;
+
+    if (!this.#isSmartTransaction) {
+      return useRegularTransactionSubmit;
```

Comment:
We don't yet have a fallback here, so could throw to have a more meaningful error than `No publish hook found` which is what we'd get currently.

Though naturally this flow shouldn't be possible at the moment.